Resync `html/rendering` from WPT Upstream

Ahmad-S792 opened the pull request:
#### d7ac06c06c6e9a22750b70a6e1978e639c43104f
<pre>
Resync `html/rendering` from WPT Upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=299061">https://bugs.webkit.org/show_bug.cgi?id=299061</a>
<a href="https://rdar.apple.com/problem/160817157">rdar://problem/160817157</a>

Reviewed by Tim Nguyen.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/40fc257a28faf7c378f59185235685ea8684e8f4">https://github.com/web-platform-tests/wpt/commit/40fc257a28faf7c378f59185235685ea8684e8f4</a>

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/dimension-attributes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/dimension-attributes.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-display-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-display.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-generated-content-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-generated-content.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-overflow.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-height-constrained-by-max-width-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-height-constrained-by-max-width-expected.xht: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-height-constrained-by-max-width.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-width-constrained-by-max-height-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-width-constrained-by-max-height-expected.xht: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-width-constrained-by-max-height.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/headingoffset-and-headingreset-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/headingoffset-and-headingreset-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/headingoffset-and-headingreset.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/WEB_FEATURES.yml:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/firefox-bug-1928736-1-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/firefox-bug-1928736-2-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/w3c-import.log:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/300255@main">https://commits.webkit.org/300255@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/43bcfacd0617d698f18040698b642ea114f5f8e6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121823 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41524 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32188 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128368 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73913 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5d4710c6-f1c3-408d-9fb4-a995b2db04d6) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42236 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50117 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92576 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61532 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/89a45c35-288c-4760-abf8-537e4d743170) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124775 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33687 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109101 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73235 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7b678b3e-a282-4950-b1f8-0c52e20a6491) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32699 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27264 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71874 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103187 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27449 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131143 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48758 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37076 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101159 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49130 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105318 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101031 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25631 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46398 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24500 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45459 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48616 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48086 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51435 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49768 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7106e99825ddad09f5e163c2ed2341eec3bd26f9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50905

Ahmad-S792 committed:
Resync `html/rendering` from WPT Upstream
https://bugs.webkit.org/show_bug.cgi?id=299061
rdar://problem/160817157

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/40fc257a28faf7c378f59185235685ea8684e8f4

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/dimension-attributes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/dimension-attributes.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-display-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-display.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-generated-content-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-generated-content.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-overflow.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-height-constrained-by-max-width-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-height-constrained-by-max-width-expected.xht: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-height-constrained-by-max-width.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-width-constrained-by-max-height-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-width-constrained-by-max-height-expected.xht: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/video-default-object-width-constrained-by-max-height.html:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/headingoffset-and-headingreset-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/headingoffset-and-headingreset-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/headingoffset-and-headingreset.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/sections/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/WEB_FEATURES.yml:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/firefox-bug-1928736-1-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/firefox-bug-1928736-2-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/w3c-import.log:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-focus-steps-expected.txt: Added.

Canonical link: https://commits.webkit.org/300255@main

webkit-commit-queue commented:
Committed 300255@main (d7ac06c06c6e): <https://commits.webkit.org/300255@main>

Reviewed commits have been landed. Closing PR #50905 and removing active labels.
